query_plan.go

```diff
@@ -237,6 +221,7 @@ func renderTreeWithStats(tree treeprint.Tree, linkType string, node *Node) {
 	statsJson, _ := protojson.Marshal(node.PlanNode.GetExecutionStats())
 	b, _ := json.Marshal(
 		map[string]interface{}{
+			"id":              fmt.Sprint(node.PlanNode.Index),
 			"execution_stats": json.RawMessage(statsJson),
 			"display_name":    node.String(),
 			"link_type":       linkType,
```

Comment:
I think using `map[string]interface{}` is not good idea as it's not statically type checked.

How about defining a type for this intermediate representation?

```go
type QueryPlanWithStats struct {
  ID string `json:"id"`
  Stats stats `json:"executionStats"`
  DisplayName string `json:"displayName"`
  LinkType string `json:"linkType"`
}

// This is for defining JSON marshal/unmarshal for protobuf
type stats struct {
  pbStats *struct.Struct
}

func (s stats) MarshalJSON() ([]byte, error) {
    ....
    protojson.Marshal(s.pbStats)
    ....
}

func (s *stats) UnmarshalJSON(data []byte) error {
    ....
    protojson.Unmarshal(data, s.pbStats)
    ....
}
```

Reply:
Fixed in https://github.com/cloudspannerecosystem/spanner-cli/pull/70/commits/2e2123fe1f91ad868c5d6641947eb5bb3c3f484e
Additionally I want to keep the type of ID field as int32 as possible.
https://github.com/cloudspannerecosystem/spanner-cli/pull/70/commits/c2a825c9d14f551ad31ff1c6d12f203278cbda24 and https://github.com/cloudspannerecosystem/spanner-cli/pull/70/commits/a3ca5e5871c4c9db9cc3e05f7f0f42834ed66806

Reply:
This type represents the information of a single operator node the name should contains `Node` or `Operator`(or `Line` or `Row`).

Reply:
Introduce `queryPlanNodeWithStatsTyped` to eliminate `getStringValueByPath` in https://github.com/cloudspannerecosystem/spanner-cli/pull/70/commits/f1919e282b8efa59a668e5062efe5c5566eba8b0